content/asp.net/getting-started/controllers-actions.md

```diff
@@ -1,20 +1,122 @@
+
+```c#
+[Route("peeps")]
+public class PersonController : Controller
+{
+    // ...
+}
+```
+
+Likewise, an action can be setup to use a different route if you apply an action to it, for instance, with the above `peeps` route set on the controller, you can have `peeps/favs` as the route for the action below with the following route applied:
+
+```c#
+[Route("peeps")]
+public class PersonController : Controller
+{
+    [Route("favs")]
+    public IActionResult Favorites()
+    {
+        //...
+    }
+}
+```
+
+In the above scenario, the action assumes the controller's route and builds upon it. You can also override the base for that specific action if you prepend the route with a forward slash:
+
+```c#
+    [Route("peeps")]
+    public class PersonController : Controller
+    {
+        [Route("favs")]
+        public IActionResult Favorites()
+        {
+            //...
+        }
+
+        [Route("/people/home")]
+        public IActionResult Index()
+        {
+            //...
+        }
+    }
+```
+
+Routes can get tricky to debug, so use them judiciously and know how they affect incoming requests in your app.  For example, someone looking at the above controller might make assumptions about the `Index` action, given the default route confiruration on the project; however, with the attributes applied as above, a request to `/person/index` would give them a `404 Not Found`.  Read more about routes [here](routing.md). 
+
+## Controller Helpers and Result Objects 
+In addition to the actions that you write yourself, you're also privy to the host of helper methods that are built into the base controller that you inherit from. These methods help to formulate a repsonse after you've completed any validation, service access or processing required. We've already looked at one as part of the samples above, but let's have another quick look:
+
+```c#
+public IActionResult Index()
+{
+    return View();
+}
+```        
+
+The call to the helper method `View` above gives you an easy way to say, "I'm done processing here, and now I would like the user to see the page related to this action."  The framework takes over, maintains the state of any properties you've set on the controller and renders a view following the default naming and location conventions we covered earlier.
+
+Just like `View`, there are other helpers that will make the lifting easier on the controller.
+
+ - **Redirection**- You can use `Redirect`, `LocalRedirect`, `RedirectToRoute` and other similar methods in your actions for the purpose of forwarding users on to other location, for example, after a login.   
+ - **Standard Response Types**- When you are working at more of an API level or need more granular control over building the response, you can use pre-built classes like `OkResult` or `NotFoundResult` to convey the results of your processing.   
```

Comment:
The helpers for these are Ok() and NotFound(). They return the result types you mention, but this is a list of helpers, right?